x-pack/plugins/fleet/server/services/epm/agent/agent.ts

```diff
@@ -136,6 +136,16 @@ function toJsonHelper(value: any) {
 }
 handlebars.registerHelper('to_json', toJsonHelper);
 
+function percentEncodeHelper(input: string) {
+    let encodedString = encodeURIComponent(input);
+    // handle left out characters
+    encodedString = encodedString
+    .replace(/[!'()*]/g, (char) => '%' + char.charCodeAt(0).toString(16).toUpperCase());
+
+    return encodedString
+}
+handlebars.registerHelper('percent_encode', percentEncodeHelper);
```

Comment:
Shouldn't we call it `url_encode`? I find a `percent_encode` not too intuitive naming.

Reply:
Renamed the helper function to `url_encode`.